addons/mail/models/discuss/discuss_channel.py

```diff
@@ -59,6 +59,9 @@ def _generate_random_token(self):
         compute='_compute_channel_partner_ids', inverse='_inverse_channel_partner_ids',
         search='_search_channel_partner_ids')
     channel_member_ids = fields.One2many('discuss.channel.member', 'channel_id', string='Members')
+    parent_channel_id = fields.Many2one("discuss.channel", help="Parent channel", ondelete="cascade", index=True)
+    sub_channel_ids = fields.One2many("discuss.channel", "parent_channel_id", string="Sub Channels")
```

Comment:
Defining (and more precisely using) this field is assuming `sub_channel_ids` is always small, otherwise we run into scaling issue and would need some kind of paging/load more.

I'm not sure how people are going to use them, but we might want to keep that in mind and do differently from the start?

Reply:
We never return the whole list. In the front-end, sub_channels are loaded lazily (pinned, or 30 by 30 in the sub-channel list).

Reply:
But as soon as the field is accessed in python, it has to fetch all of them from the db.
In the case it's ok, as the only place in non-thread where it is used, it purposely needs to get all of them.